tutorials/frontend/deploy_quantized.py

```diff
@@ -0,0 +1,154 @@
+    'http://data.mxnet.io.s3-website-us-west-1.amazonaws.com/data/val_256_q90.rec',
+    'val_256_q90.rec')
+
+def get_val_data(num_workers=4):
+    mean_rgb = [123.68, 116.779, 103.939]
+    std_rgb = [58.393, 57.12, 57.375]
+
+    def batch_fn(batch):
+        return batch.data[0].asnumpy(), batch.label[0].asnumpy()
+
+    img_size = 299 if model_name == 'inceptionv3' else 224
+    val_data = mx.io.ImageRecordIter(
+        path_imgrec=calibration_rec,
+        preprocess_threads=num_workers,
+        shuffle=False,
+        batch_size=batch_size,
+        resize=256,
+        data_shape=(3, img_size, img_size),
+        mean_r=mean_rgb[0],
+        mean_g=mean_rgb[1],
+        mean_b=mean_rgb[2],
+        std_r=std_rgb[0],
+        std_g=std_rgb[1],
+        std_b=std_rgb[2],
+    )
+    return val_data, batch_fn
+
+
+###############################################################################
+# The calibration dataset should be a iterable object. We define the
+# calibration dataset as a generator object in Python. In this tutorials, we
+# only use a few samples for calibration.
+
+calibration_samples = 10
+
+def calibrate_dataset():
+    val_data, batch_fn = get_val_data()
+    val_data.reset()
+    for i, batch in enumerate(val_data):
+        if i * batch_size >= calibration_samples:
+            break
+        data, _ = batch_fn(batch)
+        yield {'data': data}
+
+
+###############################################################################
+# Import the model
+# ----------------
+# We use the Relay MxNet frontent to import a model from the Gluon model zoo.
+def get_model():
+    gluon_model = gluon.model_zoo.vision.get_model(model_name, pretrained=True)
+    img_size = 299 if model_name == 'inceptionv3' else 224
+    data_shape = (batch_size, 3, img_size, img_size)
+    mod, params = relay.frontend.from_mxnet(gluon_model, {"data": data_shape})
+    return mod, params
+
+
+###############################################################################
+# Quantize the Model
+# ------------------
+# In quantization, we need to find the scale for each weight and output tensor.
+# For weights, the scales are directly calculated based on the value of the 
+# weights. Two modes are supported: `power2` and `max`. Both modes find the
+# maximum value within the weight tensor first. In `power2` mode, the maximum
+# is rounded down to power of two. If the scales of both weights and outputs
+# are power of two, we can leverage bit shifting for multiplications. This make
+# it computationally more efficient. In `max` mode, the maximum is used as the
+# scale. Without rounding, `max` mode might have better accuracy in some cases.
+# When the scales are not power of two, fixed point multiplications will
+# be used.
+#
+# For outputs, we can find the scales with data-aware quantization.
+# Data-aware quantization takes a calibration dataset as the input argument.
+# Scales are calculated by minimizing the KL divergence of between the data
```

Comment:
of between -> between
I'd prefer "KL divergence between distribution of activation before and after quantization"

Reply:
it is not necessarily 'activation', so i updated to 'output of each layer'

Reply:
By activation I do mean 'output of each layer' or intermediate feature map, not necessary the output of activation function like relu. I think this is a standard terminology in the literature (they talk about "weight 8bit, acitvation 32 bit" etc).